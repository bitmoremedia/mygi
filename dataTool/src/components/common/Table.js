import React from 'react';

const tableProps = {
  columns: React.PropTypes.array.isRequired,
  data: React.PropTypes.array.isRequired
}

const Table = ({columns, data}) => {

  const renderHeadings = () => {
    return columns.map((column)=><th key={`${column.key}-heading`}>{column.label}</th>);
  };

  const renderColumn = ({key, value, index})=>{
    return <td key={`${key}-${columns[index].key}`}>{value}</td>;
  };

  const renderColumns = (columnData) => {
    return columnData.map((data, index)=>renderColumn({...data, index}));
  };

  const renderRows = () => {
    return data.map((columnData)=><tr key={`${columnData[0].key}-row`}>{renderColumns(columnData)}</tr>);
  };

  return (
    <table>
      <thead>
        <tr>
          {renderHeadings()}
        </tr>
      </thead>
      <tbody>
        {renderRows()}
      </tbody>
    </table>
  );
}

Table.propTypes = tableProps;

export default Table;
