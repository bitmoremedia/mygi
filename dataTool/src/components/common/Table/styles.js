import styled from 'styled-components';

export const Container = styled.div`
  overflow-x: scroll;
  @media (min-width: 1250px) {
    overflow-x: inherit;
  }
`;
export const Table = styled.table`
  border: black 2px solid;
`;
export const TableHead = styled.thead`
`;
export const TableRow = styled.tr`
  text-align: center;
  padding-left: 20px;
  td:first-child {
    text-align: left;
  }
`;
export const TableBody = styled.tbody`
`;
export const TableHeading = styled.th`
  padding: 10px;
  border-top: 1px solid #fafafa;
  border-bottom: 1px solid #e0e0e0;
`;
export const TableColumn = styled.td`
  border-top: 1px solid #ffffff;
  border-bottom: 1px solid #e0e0e0;
  border-left: 1px solid #e0e0e0;
`;
