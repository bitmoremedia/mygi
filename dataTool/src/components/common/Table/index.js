import React, { PropTypes } from 'react';

import { Container, Table, TableHead, TableHeading, TableRow, TableBody, TableColumn } from './styles';

const TableComponent = ({columns, data}) => {

  const renderHeadings = () => {
    return columns.map((column)=><TableHeading key={`${column.key}-heading`}>{column.label}</TableHeading>);
  };

  const renderColumn = ({key, value, index})=>{
    return <TableColumn key={`${key}-${columns[index].key}`}>{value}</TableColumn>;
  };

  const renderColumns = (columnData) => {
    return columnData.map((data, index)=>renderColumn({...data, index}));
  };

  const renderRows = () => {
    return data.map((columnData)=><TableRow key={`${columnData[0].key}-row`}>{renderColumns(columnData)}</TableRow>);
  };

  return (
    <Container>
      <Table>
        <TableHead>
          <TableRow>
            {renderHeadings()}
          </TableRow>
        </TableHead>
        <TableBody>
          {renderRows()}
        </TableBody>
      </Table>
    </Container>
  );
}

TableComponent.propTypes = {
  columns: PropTypes.array.isRequired,
  data: PropTypes.array.isRequired
};

export default TableComponent;
