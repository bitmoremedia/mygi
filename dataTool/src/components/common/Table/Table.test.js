import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import Table from './index'

const columns = [
  { key: 'first-name', label: 'First Name' },
  { key: 'last-name', label: 'Last Name' },
  { key: 'age', label: 'Age' },
]
const data = [
  [
    { key: 'abcd-1', value: 'Bill' },
    { key: 'abcd-2', value: 'Oddy' },
    { key: 'abcd-3', value: 67 },
  ],
  [
    { key: 'abcde-1', value: 'John' },
    { key: 'abcde-2', value: 'Smith' },
    { key: 'abcde-3', value: 23 },
  ],
  [
    { key: 'abcdef-1', value: 'Sarah' },
    { key: 'abcdef-2', value: 'Brown' },
    { key: 'abcdef-3', value: 34 },
  ],
]

test('component renders without error', () => {
  const wrapper = mount(
    <Table columns={columns} data={data} />,
  )
  expect(wrapper.exists()).toBe(true)
})

// SNAPSHOT TESTS

test('table renders with no data', () => {
  const component = renderer.create(
    <Table columns={columns} data={[]} />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('table renders with data', () => {
  const component = renderer.create(
    <Table columns={columns} data={data} />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
