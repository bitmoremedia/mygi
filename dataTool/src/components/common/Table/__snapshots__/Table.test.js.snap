exports[`test table renders with data 1`] = `
<div
  className="eQyMHy">
  <table
    className="rLcaY">
    <thead
      className="foqgVb">
      <tr
        className="jhTQJk">
        <th
          className="jpbcPS">
          First Name
        </th>
        <th
          className="jpbcPS">
          Last Name
        </th>
        <th
          className="jpbcPS">
          Age
        </th>
      </tr>
    </thead>
    <tbody
      className="foqgVb">
      <tr
        className="jhTQJk">
        <td
          className="dqgZxL">
          Bill
        </td>
        <td
          className="dqgZxL">
          Oddy
        </td>
        <td
          className="dqgZxL">
          67
        </td>
      </tr>
      <tr
        className="jhTQJk">
        <td
          className="dqgZxL">
          John
        </td>
        <td
          className="dqgZxL">
          Smith
        </td>
        <td
          className="dqgZxL">
          23
        </td>
      </tr>
      <tr
        className="jhTQJk">
        <td
          className="dqgZxL">
          Sarah
        </td>
        <td
          className="dqgZxL">
          Brown
        </td>
        <td
          className="dqgZxL">
          34
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`test table renders with no data 1`] = `
<div
  className="eQyMHy">
  <table
    className="rLcaY">
    <thead
      className="foqgVb">
      <tr
        className="jhTQJk">
        <th
          className="jpbcPS">
          First Name
        </th>
        <th
          className="jpbcPS">
          Last Name
        </th>
        <th
          className="jpbcPS">
          Age
        </th>
      </tr>
    </thead>
    <tbody
      className="foqgVb" />
  </table>
</div>
`;
